contracts/contracts/RestrictedCalls.sol

```diff
@@ -2,25 +2,48 @@
 pragma solidity ^0.8.12;
 
 import "OpenZeppelin/openzeppelin-contracts@4.7.3/contracts/access/Ownable.sol";
+import "../interfaces/IMessenger.sol";
 
 /// A helper contract that provides a way to restrict callers of restricted functions
 /// to a single address. This allows for a trusted call chain,
 /// as described in :ref:`contracts' architecture <contracts-architecture>`.
-///
-/// .. seealso:: :sol:contract:`CrossDomainRestrictedCalls`
 contract RestrictedCalls is Ownable {
-    /// The set of callers, in the form of hashed pairs ``(chainId, address)``.
-    mapping(bytes32 => bool) public callers;
+    /// Maps caller chain IDs to tuples [caller, messenger].
+    ///
+    /// For same-chain calls, the messenger address is 0.
```

Comment:
nitpick: a better (more common) representaion forthe zero address would be 0x0 or address(0)